stages/org.osbuild.ostree.coreos.aleph

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python3
+"""
+Create aleph version file for the deployment.
+"""
+
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+from osbuild.util import containers, ostree
+
+import json
+
+CAPABILITIES = ["CAP_MAC_ADMIN"]
+
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "target_imgref": {
+      "description": "imageref used as the source of truth for updates",
+      "type": "string",
+      "pattern": "^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$",
+      "examples": ["ostree-remote-registry:fedora:quay.io/fedora/fedora-coreos:stable, ostree-image-signed:quay.io/fedora/fedora-coreos:stable, ostree-unverified-registry:quay.io/fedora/fedora-coreos:stable"]
+    },
+     "remote": {
+      "description": "optional OStree remote to use for the deployment",
+      "type": "string"
+    }
+  }
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "oneOf": [{
+    "required": ["images"]
+  }, {
+    "required": ["commits"]
+  }],
+  "properties": {
+    "images": {
+      "type": "object",
+      "description": "Container Image to deploy",
+      "additionalProperties": true
+    },
+    "commits": {
+      "type": "object",
+      "description": "OStree commit to deploy",
+      "additionalProperties": true
+    }
+  }
+}
+"""
+
+def write_to_file(tree, filename, data):
+    with open(tree + "/" + filename, "w") as f:
+        json.dump(data, f, indent=4, sort_keys=True)
+        f.write("\n")
+    f.close()
+
+    # create a symlink for backwards compatibility with CoreOS
+    cwd = os.getcwd()
+    os.chdir(tree)
+    subprocess.run(["ln", "-s", filename, ".coreos-aleph-version.json"], check=True)
+    os.chdir(cwd)
+
+
+def aleph_commit(tree, inputs, remote):
+    __, commits = ostree.parse_input_commits(inputs["commits"])
+    for commit, data in commits.items():
+        ref = data.get("ref", commit)
+
+    if remote:
+        ref = f"{remote}:{ref}"
+
+    extra_args = []
+    extra_args.append("--print-metadata-key=version")
+
+    aleph_version = ostree.show(f"{tree}/ostree/repo", "ostree/1/1/0", *extra_args)
+    aleph_ref = ref
+    aleph_ostree_commit = commit
+
+    aleph_version_data = {
+        "osbuild-version": osbuild.__version__,
+        "version": aleph_version,
+        "ref": aleph_ref,
+        "ostree-commit": aleph_ostree_commit
+    }
+
+    write_to_file(tree, ".aleph-version.json", aleph_version_data)
+
+    
+def aleph_container(tree, inputs, target_imgref):
+    images = containers.parse_containers_input(inputs)
+    for image in images.values():
+        with containers.container_source(image) as (image_name, __):
```

Comment:
is getting the `image_name` the only reason we are requiring the container to be input to this stage now? If so that is frustrating. We really want to drop requiring that input if we can.

Reply:
Yeah this was the primarily the reason why I originally put all the aleph stuff in the deploy stage. I've been trying to see if I can find the `image_name` and `target_image_name` from the deployed `ostree repo` metadata, ie:
```
ostree show --repo=/home/luyang/dev/sysroot/ostree/repo ostree/1/1/0 --list-metadata-keys
```
But I can't seem to find a good way of getting this information... maybe @cgwalters or @jlebon will have some insight on obtaining the `image name` and `target image name`?

Reply:
The target image is in the origin file.  I guess we could scrape this out, but... I dunno again my hope is we use bootc for this stuff in the future, which streamlines all of this and ties it together in an opinionated way.

Reply:
yeah that sounds good. @lukewarmtemp we should be able to pick this up from the  `/sysroot/ostree/deploy/fedora-coreos/deploy/*.0.origin` file and not have to require it as an option passed into the stage.

Reply:
@dustymabe what are your thoughts on the aleph stage taking the same inputs as the selinux stage?
```
- type: org.osbuild.ostree.selinux
  options:
    deployment:
      osname: fedora-coreos
      ref: ostree/1/1/0
```
```
- type: org.osbuild.ostree.coreos.aleph
  options:
    deployment:
      osname: fedora-coreos
      ref: ostree/1/1/0
```
The selinux stage already has a method to get the root of the system, ie: `/sysroot/ostree/deploy/fedora-coreos/deploy/*.0`

Reply:
I think that seems fine, but I also think we should make specifying it optional. i.e. if there is only one deployment we should be able to autodetect it and also autodetect the ref IMO. See https://github.com/coreos/coreos-assembler/pull/3643#discussion_r1373845109

Reply:
Sounds good, I can implement that. However, this discussion addresses the `target_imgref` problem and not the `image_name`. I still can't seem to find where the original name of the container image pulled is stored (corresponding to the `"container-image": { "image-name": <original image name>, ... }` field in the aleph file)

Reply:
I think what colin was getting at with

> The target image is in the origin file.

Is that we can just use the configured target image name here instead and that is probably a good enough approximation. i.e. the image we are using here to create images should probably also be in the repo at that name (for updating systems).